paimon-common/src/main/java/org/apache/paimon/options/CatalogOptions.java

```diff
@@ -75,4 +77,18 @@ public class CatalogOptions {
                     .defaultValue(true)
                     .withDescription(
                             "Allow to fallback to hadoop File IO when no file io found for the scheme.");
+
+    public static final Set<String> PAIMON_CATALOG_OPTIONS =
+            new HashSet<String>() {
+                {
+                    add(WAREHOUSE.key());
+                    add(METASTORE.key());
+                    add(URI.key());
+                    add(TABLE_TYPE.key());
+                    add(LOCK_ENABLED.key());
+                    add(LOCK_CHECK_MAX_SLEEP.key());
+                    add(LOCK_ACQUIRE_TIMEOUT.key());
+                    add(FS_ALLOW_HADOOP_FALLBACK.key());
+                }
```

Comment:
Can we get all fields from `CatalogOptions` instead of add them manually? Such as `FlinkConnectorOptions#getOptions`, we can add a tool class for them